frontends/main/src/page-components/Dialogs/AddToListDialog.tsx

```diff
@@ -114,13 +108,13 @@ const AddToListDialogInner: React.FC<AddToListDialogInnerProps> = ({
           })
         }
         if (listType === ListType.LearningPath) {
-          const newParents = values.learning_paths.map((id) => parseInt(id))
+          const newParents = values.learning_paths!.map((id) => parseInt(id))
           await setLearningPathRelationships({
             id: resource.id,
             learning_path_id: newParents,
           })
         } else if (listType === ListType.UserList) {
-          const newParents = values.user_lists.map((id) => parseInt(id))
+          const newParents = values.user_lists!.map((id) => parseInt(id))
```

Comment:
**Requested change:** Here and above, let's get rid of the non-null assertion. My rec instead would be to set defaults on the `user_lists` and `learning_paths`:
```ts
  const formik = useFormik({
    enableReinitialize: true,
    initialValues: {
      learning_paths: learningPathValues ?? [],
      user_lists: userListValues ?? [],
    },
    // etc ...
  })
```

_Opinion: We should not use `obj!.prop`. This has caused UI crashes in the past ([New Resource Card: null image crashes page](https://github.com/mitodl/hq/issues/4530#top)#4530) that would have been caught by TS had we not used a non-null assertion._

### Additionally
I think the checkboxes should be disabled while membership info is still loading. This should be as simple as saying
```
<CheckboxChoiceField disabled={
  !userListValues /* or maybe grab isLoading off the query*/
} />
```
but apparently `CheckboxChoiceField` doesn't have a disabled prop. Adding one should be easy—just a matter of passing the prop to the underlying html checkbox.

Reply:
I was reluctant to disable checkboxes here as these lists can never not be loaded by the time we open the dialog -  the membership lists will have been fetched to fill the resource card action buttons; these hooks just select the value for the resource. I added for good measure as we could in theory use the dialog from some other entrypoint in the future, plus the checkboxes do need to be disableable (is that a word?) generally for shared use.

Reply:
> as these lists can never not be loaded by the time

In principle they can be—if the membership endpoint response is slower than others for some reason.

> plus the checkboxes do need to be disableable

👍 👍